packages/moonstone/Panels/Panels.js

```diff
@@ -253,18 +231,23 @@ const PanelsBase = kind({
 		viewportId: ({id}) => id && `${id}-viewport`
 	},
 
-	render: ({arranger, childProps, children, controls, generateId, id, index, noAnimation, noSharedState, viewportId, ...rest}) => {
-		delete rest.closeButtonBackgroundOpacity;
-		delete rest.closeButtonAriaLabel;
+	render: ({arranger, childProps, children, closeButtonAriaLabel, closeButtonBackgroundOpacity, controls, controlsRef, generateId, id, index, noAnimation, noCloseButton, noSharedState, onApplicationClose, viewportId, ...rest}) => {
 		delete rest.controlsMeasurements;
-		delete rest.controlsRef;
-		delete rest.noCloseButton;
-		delete rest.onApplicationClose;
 		delete rest.onBack;
 
 		return (
 			<div {...rest} id={id}>
-				{controls}
+				<Controls
+					closeButtonAriaLabel={closeButtonAriaLabel}
+					closeButtonBackgroundOpacity={closeButtonBackgroundOpacity}
+					id={getControlsId(id)}
```

Comment:
If this value is used twice, can it be a computed prop or calculated at the top of the render function?  Would it save much performance-wise?

Reply:
It's used both in a computed prop and the render so it itself can't be in computed since each of those is evaluated independent of each other by design. We could memoize it but I doubt we'd buy ourselves much in perf for string concatenation.

Reply:
I meant it's getting used twice in these props for `id` and `spotlightId`.  I have this thing about calling the function once :lol:

Reply:
ah, right! I originally had some other code in here to change spotlight to use `id` as a fallback for `spotlightId` but decided to simplify this PR and remove that. I'll move this up to the top of the fn. Thanks for the catch and explanation.